contracts/src/v0.8/dev/functions/FunctionsOracle.sol

```diff
@@ -76,6 +77,48 @@ contract FunctionsOracle is FunctionsOracleInterface, OCR2Base, AuthorizedOrigin
     s_donPublicKey = donPublicKey;
   }
 
+  /**
+   * @dev check if node is in current transmitter list
+   */
+  function _isTransmitter(address node) internal view returns (bool) {
+    address[] memory nodes = this.transmitters();
+    for (uint i = 0; i < nodes.length; i++) {
+      if (nodes[i] == node) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * @inheritdoc FunctionsOracleInterface
+   */
+  function setNodePublicKey(address node, bytes calldata publicKey) external {
+    // Owner can set anything. Transmitters can set only their own key.
+    require(msg.sender == owner() || (_isTransmitter(msg.sender) && msg.sender == node), "unauthorized public key change");
+    s_nodePublicKeys[node] = publicKey;
+  }
+
+  /**
+   * @inheritdoc FunctionsOracleInterface
+   */
+  function deleteNodePublicKey(address node) external {
```

Comment:
This will be effectively equal to `setNodePublicKey(node, 0)`.

Reply:
Yes but I guess it's still cleaner to keep both.